src/utilities.jl

```diff
@@ -139,6 +139,52 @@ function peekchar(io::(isdefined(Base, :GenericIOBuffer) ?
     return convert(Char,c)
 end
 
+function dpeekchar(io::IOBuffer)
```

Comment:
Isn't it possible to cut the size of this function in roughly a half since the two code blocks only differ by a `+offset`(perhaps factor out the block in a new function and call it twice)? Also, could run your code formatter on this since it lacks some spaces between arguments.

Reply:
I've refactored things a bit, adding a `utfread` function. The formatter is a bit broken (I'll have a look aat the moment so I've made some changes